### PR TITLE
morpc: multi-streams do not reuse same connection

### PR DESCRIPTION
**Pull request by zhangxu19830126**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6397 

## What this PR does / why we need it:
Currently, morpc's streaming implementation makes multiple streams maybe reuse the same tcp connection. However, under this bug, Pipeline's will depend on each other, and due to the low concurrency, the streams of these Pipelines use the same TCP connection, which causes some Pipelines not to process the messages received by the streaming when they do not receive some responses, resulting in the receive buffer of the corresponding tcp at the bottom being written The receiving buffer of the underlying corresponding tcp is written full, and the server will be blocked when sending messages through this connection, which leads to other pipelines not receiving messages either, and eventually leads to the whole process being stuck.

### CODE REVIEW

**Review by nnsgmsone — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Review by yingfeng — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

File: pkg/common/morpc/client.go [hunk 1]
```diff
@@ -235,7 +235,7 @@ func (c *client) getBackendLocked(backend string) (Backend, error) {
 		for i := uint64(0); i < n; i++ {
 			seq := c.mu.ops[backend].next()
 			b = backends[seq%n]
-			if b.LastActiveTime() != (time.Time{}) {
+			if !b.Locked() && b.LastActiveTime() != (time.Time{}) {
```
Comment (fengttt) [inline pkg/common/morpc/client.go:238]: I don't thik this code is correct.   Seems two clients, calling getBackendLocked, will get same b.

Later line 251 will crash.

Comment (fengttt) [reply pkg/common/morpc/client.go:238]: Never mind, you called c.mu.Lock before calling this, correct?   So far away ...

File: pkg/common/morpc/client.go [hunk 2]
```diff
@@ -378,19 +382,26 @@ func (c *client) createTask(ctx context.Context) {
 	}
 }
 
-func (c *client) createBackend(backend string) (Backend, error) {
+func (c *client) createBackend(backend string, lock bool) (Backend, error) {
```
Comment (fengttt) [inline pkg/common/morpc/client.go:385]: Is this function doing same thing as getBackend?     Should we merge these as same function?

**Comment by fengttt** [comment]
@mergifyio refresh

File: pkg/common/morpc/client.go (hunk 1 above)
Comment (zhangxu19830126) [reply pkg/common/morpc/client.go:238]: no, two clients cannot call getBackendLocked at the same time. It's protected by s.mu.Lock

File: pkg/common/morpc/client.go (hunk 2 above)
Comment (zhangxu19830126) [reply pkg/common/morpc/client.go:385]: getBackend tries to get an available backend to return to the client,  and if none is available it calls createBackend to create one.

**Review by fengttt — APPROVED** [review]

**Comment by fengttt** [comment]
@mergifyio refresh

**Pull request merged**